backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -101,65 +105,78 @@ public CDateRange getEntityDateRange(int entity) {
 	}
 
 	public int getEntityMinDate(int entity) {
-		return minDate.getOrDefault(entity, Integer.MIN_VALUE);
+		return minDate[getEntityIndex(entity)];
 	}
 
 	public int getEntityMaxDate(int entity) {
-		return maxDate.getOrDefault(entity, Integer.MAX_VALUE);
+		return maxDate[getEntityIndex(entity)];
 	}
 
 	@Override
 	@JsonIgnore
 	public CBlockId createId() {
-		return new CBlockId(bucket, connector);
+		return new CBlockId(bucket.getId(), connector);
 	}
 
 	public void initIndizes(int bucketSize) {
-		includedConcepts = new Int2LongArrayMap(bucketSize);
-		includedConcepts.defaultReturnValue(0);
+		root = bucket.getBucket() * bucketSize;
+
+		includedConcepts = new long[bucketSize];
+
+		minDate = new int[bucketSize];
+		maxDate = new int[bucketSize];
 
-		minDate = new Int2IntArrayMap(bucketSize);
-		maxDate = new Int2IntArrayMap(bucketSize);
+		for (int index = 0; index < bucketSize; index++) {
+			includedConcepts[index] = 0;
+
+			minDate[index] = Integer.MIN_VALUE;
+			maxDate[index] = Integer.MAX_VALUE;
+		}
 	}
 
 	public void addEntityDateRange(int entity, CDateRange range) {
+		final int index = getEntityIndex(entity);
+
 		if (range.hasLowerBound()) {
 			final int minValue = range.getMinValue();
 
-			if (!minDate.containsKey(entity)) {
-				minDate.put(entity, minValue);
+			if (minDate[index] != Integer.MIN_VALUE) {
+				minDate[index] = minValue;
```

Comment:
```suggestion
			if (minDate[index] == Integer.MIN_VALUE) {
				minDate[index] = minValue;
```

oder sehe ich das falsch?

Reply:
sonst ist der else branch immer MIN_VALUE